featuretools/tests/entityset_tests/test_es.py

```diff
@@ -1392,15 +1415,12 @@ def test_normalize_with_numeric_time_index(int_es):
 
 def test_normalize_with_invalid_time_index(es):
     es['customers'].convert_variable_type('signup_date', variable_types.Datetime)
-    error_text = "Time index 'signup_date' is not a NumericTimeIndex or DatetimeTimeIndex," \
-        + " but type <class 'featuretools.variable_types.variable.Datetime'>."\
-        + " Use set_time_index on entity 'customers' to set the time_index."
-    with pytest.raises(TypeError, match=error_text):
-        es.normalize_entity(base_entity_id="customers",
-                            new_entity_id="cancel_reason",
-                            index="cancel_reason",
-                            copy_variables=['upgrade_date'])
-    es['customers'].convert_variable_type('signup_date', variable_types.DatetimeTimeIndex)
+    es.normalize_entity(base_entity_id="customers",
+                        new_entity_id="cancel_reason",
+                        index="cancel_reason",
+                        copy_variables=['upgrade_date'])
+
+    assert isinstance(es['customers']['signup_date'], variable_types.DatetimeTimeIndex)
```

Comment:
There is a subtle difference. In the previous behavior, the variable type would get changed in `set_time_index`. If a user converts the variable type again after, it will be invalid when needed in `normalize_entity`. In this behavior, the variable type is not changed until it's needed in`normalize_entity`.